impl/client/src/doc/javascript/template/vizApi/tmpl/type.tmpl

```diff
@@ -1,7 +1,207 @@
 <?js
-    var data = obj;
-    var self = this;
-    data.forEach(function(name, i) { ?>
-<span class="param-type"><?js= self.linkto(name, self.htmlsafe(name)) ?></span>
-<?js if (i < data.length-1) { ?>|<?js } ?>
-<?js }); ?>
\ No newline at end of file
+  var data = obj;
+  var self = this;
+  var mdnJsTypes = {
+    "array": "Array",
+    "boolean": "Boolean",
+    "date": "Date",
+    "error": "Error",
+    "typeerror": "TypeError",
+    "function": "Function",
+    "infinity": "Infinity",
+    "json": "JSON",
+    "nan": "NaN",
+    "null": "null",
+    "number": "Number",
+    "object": "Object",
+    "string": "String",
+    "promise": "Promise",
+    "undefined": "undefined"};
+  var mdnJsTypeBaseURL = "http://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/";
+  var backboneTypeBaseURL = "http://backbonejs.org/#";
+  var jQueryTypes = {
+    "jquery": "jQuery",
+    "selector": "Selector"
+  }
+  var jQueryTypeBaseURL = "http://api.jquery.com/Types/#";
+
+  var modifiers = "[?!]?";
+  var left_p = "\\(?";
+  var right_p = "\\)?";
+  var remainder = "$|(?:[|,]\\s*)(.+)";
+  var complexProps = "[<(?!\\w|,. )>]+";
+  var BACKBONE_TYPE_REGX = /backbone\.([a-z]+)/i;
+
+  // regx: /([?!]?)([\w.*]+)(?:())?/
+  var SIMPLE_TYPE_REGX = new RegExp("(" + modifiers + ")([\\w.*]+)(?:\\(\\))?");
```

Comment:
You still don't provide any examples of the several cases the regexes should match. This will be invaluable in the future.